packages/web-pkg/src/components/AppTopBar.vue

```diff
@@ -1,24 +1,52 @@
 <template>
-  <div class="oc-flex oc-p-s app-top-bar">
-    <oc-resource
-      v-if="resource"
-      id="app-top-bar-resource"
-      :is-thumbnail-displayed="false"
-      :resource="resource"
-    />
-    <div v-else />
-    <div>
-      <slot name="right" />
-      <oc-button
-        id="app-top-bar-close"
-        :aria-label="$gettext('Close')"
-        size="small"
-        @click="$emit('close')"
-      >
-        <oc-icon name="close" size="small" />
-      </oc-button>
+  <portal to="app.runtime.header.left">
+    <div class="oc-app-top-bar oc-p-xs oc-flex oc-flex-center">
+      <div class="open-file-bar oc-flex oc-mr-s">
+        <oc-resource
+          v-if="resourceReady"
+          id="open-file-bar-resource"
+          :is-thumbnail-displayed="false"
+          :resource="resource"
+        />
+        <oc-resource v-else :resource="{ name: 'Loading' }" />
+        <template v-if="hasDropDownActions">
+          <oc-button
+            id="oc-openfile-contextmenu-trigger"
+            v-oc-tooltip="contextMenuLabel"
+            :aria-label="contextMenuLabel"
+            appearance="raw"
+          >
+            <oc-icon name="more-2" />
+          </oc-button>
+
+          <oc-drop
+            drop-id="oc-openfile-contextmenu"
+            toggle="#oc-openfile-contextmenu-trigger"
+            mode="click"
+            close-on-click
+            padding-size="small"
+            @click.stop.prevent
+          >
+            <slot name="contextActions" />
+          </oc-drop>
+        </template>
+      </div>
+
+      <div class="oc-flex main-actions oc-my-s">
+        <slot name="mainActions" />
```

Comment:
@kulmann slots or creating a list by looping over an array of a clearly defined interface/type?

Reply:
Same as above, even the same type for the prop, but different prop. :-)